tap_facebook/__init__.py

```diff
@@ -103,16 +103,25 @@ def iter_delivery_info_filter(stream_type):
         yield filt
 
 def retry_pattern(backoff_type, exception, **wait_gen_kwargs):
+    # HACK: Workaround added due to bug with Facebook prematurely deprecating 'relevance_score'
+    # Issue being tracked here: https://developers.facebook.com/support/bugs/2489592517771422
+    def is_relevance_score(exception):
+        return exception.body()["error"]["message"] == '(#100) relevance_score is not valid for fields param. please check https://developers.facebook.com/docs/marketing-api/reference/ads-insights/ for all valid values'
+
     def log_retry_attempt(details):
         _, exception, _ = sys.exc_info()
+        if is_relevance_score(exception):
+            LOGGER.info("Detected that 'relevance_score' is not available, removing from Insights export and continuing.")
+            details['args'][1]['fields'].remove('relevance_score')
```

Comment:
I actually think the safest strategy here would be to raise an exception with something like the following text:

```
Due to bug with Facebook prematurely deprecating 'relevance_score' that is
not affecting all tap-facebook clients in the same way, you need to
deselect `relevance_score` from your Insights export. For further
information, please see this active Facebook bug:
https://developers.facebook.com/support/bugs/2489592517771422.
```

If we did that instead of proactively deselecting the field, I think we're keeping the locus of control where it should be (in the customers hand), preventing hidden data discrepancy conversations, and also pointing blame to where it's due in a very visible way.

Reply:
I like that approach better. I checked the ticket today, and it seems like there's been no progress (over the weekend, so that makes sense). If it makes sense to merge this, I'll change it to use that pattern.